netbox/extras/scripts_worker.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+import requests
+from core.models import Job
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth import get_user_model
+from django_rq import job
+from jinja2.exceptions import TemplateError
+from utilities.rqworker import get_workers_for_queue
+
+from extras.constants import WEBHOOK_EVENT_TYPES
+from extras.models import ScriptModule
+from extras.scripts import run_script
+from extras.utils import eval_conditions
+from extras.webhooks import generate_signature
+
+logger = logging.getLogger('netbox.webhooks_worker')
+
+
+@job('default')
+def process_script(event_rule, model_name, event, data, timestamp, username, request_id=None, snapshots=None):
+    """
+    Run the requested script
+    """
+    if not eval_conditions(event_rule, data):
+        return
+
+    module_id = event_rule.action_parameters.split(":")[0]
+    script_name = event_rule.action_parameters.split(":")[1]
+
+    try:
+        module = ScriptModule.objects.get(pk=module_id)
+    except ScriptModule.DoesNotExist:
+        logger.warning(f"event run script - script module_id: {module_id} script_name: {script_name}")
+        return
+
+    try:
+        user = get_user_model().objects.get(username=username)
+    except ObjectDoesNotExist:
+        logger.warning(f"event run script - user does not exist username: {username} script_name: {script_name}")
+        return
```

Comment:
Would a more appropriate course of action be to set the user to None and proceed with executing the script?

Reply:
As per discussion, not sure on this one - I'm wondering if we want to create a job and have it error, then it is easy for the admin to see the error.  This way the script is never executed and you have to look at the error logs to figure out why.